dts/arm/st/pinctrl_st_stm32.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020 Linaro Limited
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef PINCTRL_ST_STM32_H_
+#define PINCTRL_ST_STM32_H_
+
+/* Create a pincfg device tree node:
+ *
+ * The node name and nodelabel will be of the form:
+ *
+ * NODE = <periph>_<signal>_p<port>_<pin>
+ *
+ * LABEL: NODE {
+ *     pinmux = <STM32_PINMUX_Z(STM32_DT_PORT_<port>,pin, function)>;
+ *     prop-foo;
+ *     prop-bar;
+ * }
+ *
+ * Also, each macro will fill directly a knwon working pin configuration
+ * depending on the signal that is required.
+ *
+ * So for example:
+ *
+ * DT_STM32_UART_TX(b, 6, usart1, AF7);
+ *
+ * Will become:
+ *
+ * usart1_tx_pb6: usart1_tx_pb6 {
+ *     pinmux = <((((0x10 + 6)) << 8) | (0x7))>;
+ *     drive-push-pull;
+ *     bias-pull-up;
+ * };
+ *
+ */
+
+#define DT_STM32_UART_TX(port, pin, periph, function) \
```

Comment:
Macros should really be documented as they not only set the pinmux but also the defaut pin configurations (e.g. drive-push-pull, bias-pull-up). 

IMHO using a script to generate what DT_STM32_UART_xx currently does would make the default pin configuration more obvious.

Reply:
> Macros should really be documented as they not only set the pinmux but also the defaut pin configurations (e.g. drive-push-pull, bias-pull-up).
> 
Sure, they need to be documented.

> IMHO using a script to generate what DT_STM32_UART_xx currently does would make the default pin configuration more obvious.
I'm not sure to get this point. Can you elaborate ?

Reply:
>> IMHO using a script to generate what DT_STM32_UART_xx currently does would make the default pin configuration more obvious.

> I'm not sure to get this point. Can you elaborate ?

Using a script to generate the xxx-pinctrl.dtsi files could generate

```
      usart1_tx_pb6: usart1_tx_pb6 {
          pinmux = <STM32_PINMUX_Z(STM32_DT_PORT_b, 6, AF7)>;
          drive-push-pull;
          bias-pull-up;
      };
     ...
```

instead of

```
     DT_STM32_UART_TX(usart1,b,6,AF7);
     ...
```

Reply:
I think that the script can afford the generation of the direct entries (i.e. no macros). Script will just need to contain the drive/bias defaults for each peripheral type. I suppose the original idea of the macros was to simplify maintenance of pinctrl files, but now that their generation is automated adding new models takes just a few seconds.

Reply:
This is indeed a valid alternative which as an advantage in readability (and solution for case issue).
However this would end up generating much larger files and would require a (bit) more complex generation mechanism.
This should be discussed.

Reply:
@erwango regarding script changes they are easy, I can change it to generate direct entries

Reply:
Using STM32_PINMUX_Z(STM32_DT_PORT_b, 6, AF7) adds some driver flexibility. Driver changes can be done without affecting all the xxx-pinctrl.dtsi files and the generation script. It´s a trade off like always.

Reply:
Ok so remains the size of the generated output. 
Using macros we're already around 2M today, direct entries would be around 3x times more (tbc) and that's for 4 signals. Considering we'd have around 15 signals average (tbc), we'd end up with less of 30M, which seems actually reasonable.
We can have a try...